apps/src/storage/levelbuilder/Dataset.jsx

```diff
@@ -14,17 +37,58 @@ class Dataset extends React.Component {
     // it as an array."
     // https://firebase.googleblog.com/2014/04/best-practices-arrays-in-firebase.html
     tableRecords: PropTypes.oneOfType([PropTypes.object, PropTypes.array])
-      .isRequired
+      .isRequired,
+    onUploadComplete: PropTypes.func.isRequired
+  };
+
+  state = {
+    notice: null,
+    isError: false
+  };
+
+  displayNotice = (notice, isError) => {
+    this.setState({notice, isError}, () =>
+      setTimeout(() => this.setState({notice: null, isError: false}), 5000)
+    );
+    window.scrollTo(0, 0);
   };
+
+  importCsv = (csv, onComplete) => {
+    $.ajax({
+      url: `/datasets/${this.props.tableName}`,
+      method: 'PUT',
+      contentType: 'application/json',
+      data: JSON.stringify({csv_data: csv})
+    })
+      .done(data => {
+        this.props.onUploadComplete(
+          this.props.tableName,
+          Object.values(data.records),
+          data.columns
+        );
+        this.displayNotice('Upload succeeeded.', false);
+        onComplete();
+      })
+      .fail(() => this.displayNotice('Upload failed.', false));
```

Comment:
Should `false` be `true` here? (Also, can you include a comment for the bool literal here?)